### PR TITLE
Add begin to render requests to a4a ad

### PR DESCRIPTION
**Pull request by zhouyx**
As agreed in cr/325220403

The the response now also support `btrUrl` as an optional array.
The begin to render config looks like
```
{
  on: ini-load,
  selector: amp-ad,
  selectionMethod: closest
}
```

### CODE REVIEW

File: ads/google/a4a/utils.js [hunk 1]
```diff
@@ -707,6 +708,23 @@ export function extractAmpAnalyticsConfig(a4a, responseHeaders) {
     // Security review needed here.
     config['requests'] = requests;
     config['triggers']['continuousVisible']['request'] = Object.keys(requests);
+
+    // Add begin to render requests
+    const btrUrls = analyticsConfig['btrUrl'];
+    if (Array.isArray(btrUrls) && btrUrls.length) {
+      const btrRequests = dict();
+      for (let idx = 1; idx <= btrUrls.length; idx++) {
```
Comment (jeffkaufman) [inline ads/google/a4a/utils.js:716]: nit: I would find this very slightly clearer if you iterated over `btrUrls` in the ordinary (zero indexed) fashion, and then referred to `btrRequests['btr${idx+1}']`

Comment (zhouyx) [reply ads/google/a4a/utils.js]: I agree. Switched to zero index.

File: ads/google/a4a/utils.js [hunk 2]
```diff
@@ -707,6 +708,23 @@ export function extractAmpAnalyticsConfig(a4a, responseHeaders) {
     // Security review needed here.
     config['requests'] = requests;
     config['triggers']['continuousVisible']['request'] = Object.keys(requests);
+
+    // Add begin to render requests
+    const btrUrls = analyticsConfig['btrUrl'];
```
Comment (tedherbstman) [inline ads/google/a4a/utils.js:713]: if analyticsConfig['url'] on line 610 is non existent, then we will never get here, right?

Comment (tedherbstman) [reply ads/google/a4a/utils.js:713]: sorry line 682

Comment (zhouyx) [reply ads/google/a4a/utils.js:713]: Sync offline. We want to handle url and btrUrl separately. In theory an invalid or empty url response should not affect the begin to render request.

Comment (zhouyx) [reply ads/google/a4a/utils.js]: fixed. Proceed with `btrUrl` with invalid or empty `url` response

**Comment by zhouyx** [comment]
@lannka could you take a look. Thanks

**Pull request merged**